examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,287 @@
+# the other. (The inverse partition, :math:`z=0101` is of course equally valid.) In the following sections,
+# we will represent partitions using computational basis states and use PennyLane to
+# rediscover this optimal cut.
+#
+# A quantum circuit
+# ~~~~~~~~~~~~~~~~~
+# This section presents the operators used in the QAOA algorithm and their implementation using basic unitary gates.
+# Firstly, if the partition is given by the computational basis states, we can represent the terms in the
+# objective function as operators like so
+#
+# .. math::
+#   C_\alpha = \frac{1}{2}\left(1-\sigma_{z}^j\sigma_{z}^k\right)
+#
+# where the :math:`\alpha\text{th}` edge is between qubits/vertices :math:`(j,k)`.
+# :math:`C_\alpha` is 1 if and only if the :math:`j\text{th}` and :math:`k\text{th}`
+# qubits have different z-axis measurement values, representing separate partitions.
+# The objective function :math:`C` is now a diagonal operator with integer eigenvalues.
+#
+# QAOA starts with a uniform superposition over the :math:`n` bitstring basis states,
+#
+# .. math::
+#   |+_{n}\rangle = \frac{1}{\sqrt{2^n}}\sum_{z=0}^{2^n-1} |z\rangle
+#
+#
+# We aim to explore the space of bitstring states for a superposition which is likely to yield a
+# large value for the :math:`C` operator upon performing a measurement in the computational basis.
+# Using the :math:`2p` angle parameters
+# :math:`\boldsymbol{\gamma} = \gamma_1\gamma_2...\gamma_p`, :math:`\boldsymbol{\beta} = \beta_1\beta_2...\beta_p`
+# we perform a sequence of operations on our initial state:
+#
+# .. math::
+#   |\boldsymbol{\gamma},\boldsymbol{\beta}\rangle = U_{B_p}U_{C_p}U_{B_{p-1}}U_{C_{p-1}}...U_{B_1}U_{C_1}|+_n\rangle
+#
+# where the operators have the explicit forms
+#
+# .. math::
+#   U_{B_i} &= e^{-i\beta_iB} = \prod_{j=1}^n e^{-i\beta_i\sigma_x^j} \\
+#   U_{C_i} &= e^{-i\gamma_iC} = \prod_{\text{edge (j,k)}} e^{-i\gamma_i(1-\sigma_z^j\sigma_z^k)/2}
+#
+# In other words, we make :math:`p` layers of our parametrized :math:`U_bU_C` gates.
+# These can be implemented on a quantum circuit using the gates depicted below (up to an irrelevant constant
+# that gets absorbed into the parameters).
+#
+# .. figure:: ../../examples/figures/qaoa_operators.png
+#    :align: center
+#    :scale: 100%
+#    :alt: qaoa_operators
+#
+# |
+# Let :math:`\langle \boldsymbol{\gamma},
+# \boldsymbol{\beta} | C | \boldsymbol{\gamma},\boldsymbol{\beta} \rangle` be the expectation of our objective.
+# In the next section, we will use PennyLane to sample this expectation value and perform classical optimization
+# over the parameters. This will specify a state :math:`|\boldsymbol{\gamma},\boldsymbol{\beta}\rangle` which is
+# likely to yield an approximately optimal partition :math:`|z\rangle` upon performing a measurement in the
+# computational basis.
+# In the case of the graph shown above, we want to measure either 0101 or 1010 from our state.
```

Comment:
```suggestion
# In the case of the graph shown above, we want to measure either 0101 or 1010 from our state, since these correspond to the optimal partitions.
```